Algorithm/QCAlgorithm.Indicators.cs

```diff
@@ -305,6 +305,29 @@ public AverageTrueRange ATR(Symbol symbol, int period, MovingAverageType type =
             return averageTrueRange;
         }
 
+        /// <summary>
+        /// Creates an AugenPriceSpike indicator for the symbol. The indicator will be automatically
+        /// updated on the given resolution.
+        /// </summary>
+        /// <param name="symbol">The symbol whose APS we want</param>
+        /// <param name="period">The period of the APS</param>
+        /// <param name="resolution">The resolution</param>
+        /// <param name="selector">Selects a value from the BaseData to send into the indicator, if null defaults to the Value property of BaseData (x => x.Value)</param>
+        /// <returns>The AugenPriceSpike indicator for the given parameters</returns>
+        public AugenPriceSpike APS(Symbol symbol, int period = 2, Resolution? resolution = null, Func<IBaseData, decimal> selector = null)
+        {
+            var name = CreateIndicatorName(symbol, $"APS({period})", resolution);
+            var augenPriceSpike = new AugenPriceSpike(name, period);
+            RegisterIndicator(symbol, augenPriceSpike, resolution, selector);
+
+            if (EnableAutomaticIndicatorWarmUp)
+            {
+                WarmUpIndicator(symbol, augenPriceSpike, resolution);
+            }
+
+            return augenPriceSpike;
```

Comment:
Cool!
These 5 lines are repeated so many times 😄, I bet they can be reduced with some private method